src/service.rs

```diff
@@ -108,59 +108,54 @@ impl Account {
     pub fn signup(client: Client, user: &User, password: &str) -> Result<Self> {
         super::init()?;
 
-        let salt = randombytes(32);
+        let salt = randombytes_array();
```

Comment:
`derive_key` only ever used the first 16 bytes of the salt, and now only 16 bytes are generated instead of 32. I don't think that's a problem?

Reply:
I think we were using the other 16 bytes for something else. Though anyhow, that's what the server expects and what we've been doing all over. Let's maintain this and send the server 16bytes?

Reply:
Assuming you meant "send the server 32bytes" - sure, I'll revert it to that.

Reply:
Yeah, that's what I meant. :)